lib/iris/_concatenate.py

```diff
@@ -433,10 +452,14 @@ def match(self, other, error_on_mismatch):
 
         # Check cube definitions.
         if self.defn != other.defn:
-            # Note that the case of different phenomenon names is dealt with
-            # in :meth:`iris.cube.CubeList.concatenate_cube()`.
-            msg = 'Cube metadata differs for phenomenon: {}'
-            msgs.append(msg.format(self.defn.name()))
+            promoted = self.promote_defn()
```

Comment:
Explain here too.  Something like "Check whether lazy versions of the defns would match."